narwhals/_spark_like/dataframe.py

```diff
@@ -173,16 +176,87 @@ def sort(
 
         flat_by = flatten([*flatten([by]), *more_by])
         if isinstance(descending, bool):
-            descending = [descending]
+            descending = [descending] * len(flat_by)
 
         if nulls_last:
-            sort_funcs = [
+            sort_funcs = (
                 F.desc_nulls_last if d else F.asc_nulls_last for d in descending
-            ]
+            )
         else:
-            sort_funcs = [
+            sort_funcs = (
                 F.desc_nulls_first if d else F.asc_nulls_first for d in descending
-            ]
+            )
 
         sort_cols = [sort_f(col) for col, sort_f in zip(flat_by, sort_funcs)]
         return self._from_native_frame(self._native_frame.sort(*sort_cols))
+
+    def with_row_index(self: Self, name: str) -> Self:
+        import pyspark.sql.functions as F  # noqa: N812
+
+        return self._from_native_frame(
+            self._native_frame.withColumn(name, F.monotonically_increasing_id())
+        )
+
+    def drop_nulls(self: Self, subset: str | list[str] | None) -> Self:
+        return self._from_native_frame(self._native_frame.dropna(subset=subset))
+
+    def rename(self: Self, mapping: dict[str, str]) -> Self:
+        return self._from_native_frame(
+            self._native_frame.withColumnsRenamed(colsMap=mapping)
+        )
+
+    def unique(
+        self: Self,
+        subset: str | list[str] | None = None,
+        *,
+        keep: Literal["any", "first", "last", "none"],
+        maintain_order: bool,
+    ) -> Self:
+        if keep != "any":
+            msg = "`LazyFrame.unique` with PySpark backend only supports `keep='any'`."
+            raise ValueError(msg)
+
+        if maintain_order is True:
+            warnings.warn(
+                "Argument `maintain_order=True` is unused in `LazyFrame.unique` with "
+                "PySpark backend",
+                UserWarning,
+                stacklevel=find_stacklevel(),
+            )
```

Comment:
we should probably raise here, if it's unused?

Reply:
In dask `maintain_order` is unused as well, without any warnings 🤔